src/MoBi.Core/Domain/Services/SimulationFactory.cs

```diff
@@ -94,25 +96,25 @@ private void validateDimensions(IModel model, SimulationBuilder simulationBuilde
             .SecureContinueWith(t => showWarnings(t.Result));
       }
 
-      public IModel CreateModelAndValidate(SimulationConfiguration simulationConfiguration, string modelName, string message = AppConstants.Captions.ConfiguringSimulation)
+      public CreationResult CreateModelAndValidate(SimulationConfiguration simulationConfiguration, string modelName, string message = AppConstants.Captions.ConfiguringSimulation)
       {
          CreationResult results = null;
 
-         _heavyWorkManager.Start(() => { results = createModel(simulationConfiguration, modelName); }, message);
+         _heavyWorkManager.Start(() => results = createModel(simulationConfiguration, modelName), message);
 
          if (results == null || results.IsInvalid)
             throw new MoBiException(AppConstants.Exceptions.CouldNotCreateSimulation);
 
          validateDimensions(results.Model, results.SimulationBuilder);
 
-         return results.Model;
+         return results;
       }
 
       public IMoBiSimulation CreateSimulationAndValidate(SimulationConfiguration configurationReferencingBuildingBlocks, string simulationName)
       {
-         var model = CreateModelAndValidate(configurationReferencingBuildingBlocks, simulationName, AppConstants.Captions.CreatingSimulation);
+         var results = CreateModelAndValidate(configurationReferencingBuildingBlocks, simulationName, AppConstants.Captions.CreatingSimulation);
          var clonedConfiguration = _cloneManager.Clone(configurationReferencingBuildingBlocks);
-         return CreateFrom(clonedConfiguration, model).WithName(simulationName);
+         return CreateFrom(clonedConfiguration, results.Model, results.SimulationBuilder.EntitySources).WithName(simulationName);
```

Comment:
I don;t think what I did is going to work at all with the way it's implemented. I need help to understand why it's done like this
As you can see, we are NOT using the configuration passed as a parameter but a clone. That means that the id of building block and quantity will ALL be different. So my sources will be pointing to the cloned version and not to the original version.. Maybe it's ok, maybe it's not? no idea